apps/authentication-service/src/shared/users/usersService.test.ts

```diff
@@ -0,0 +1,61 @@
+import { http, HttpResponse } from "msw";
+
+import { User } from "@/shared/users/usersModel";
+import { AUTH0_USER_BY_ID } from "@/test/handlers";
+import {
+  exampleClientID,
+  exampleUserAlreadyHasAClientError,
+  exampleUserID,
+  exampleUserInfoResponse,
+  exampleUserWithClientId,
+} from "@/test/testData/users";
+import { server } from "@/test/testServer";
+import { getUserClientId, getUserById, setUserClientId } from "./usersService";
+
+// eslint-disable-next-line @typescript-eslint/no-unsafe-return
+jest.mock("auth0", () => ({
+  ...jest.requireActual("auth0"),
+  UserInfoClient: class {
+    getUserInfo = (_token: string) => ({ data: exampleUserInfoResponse });
+  },
+}));
+
+describe("User Service tests", () => {
+  describe("getUserById", () => {
+    it("returns a user resource", async () => {
+      const user: User = await getUserById(exampleUserID);
+      expect(user).toEqual(exampleUserWithClientId);
+    });
+  });
+
+  describe("getUserClientId", () => {
+    it("gets the client id from a user resource", async () => {
+      const clientId = await getUserClientId(exampleUserID);
+      expect(clientId).toBe(exampleClientID);
+    });
+  });
+
+  describe("setUserClientId", () => {
+    it("sets a user's client id", async () => {
+      const user = await setUserClientId(exampleUserID, exampleClientID);
+      expect(user).toEqual(exampleUserWithClientId);
+    });
+  });
+
+  describe("setUserClientId errors", () => {
+    it("handles an api failure", async () => {
+      server.use(
+        http.patch(AUTH0_USER_BY_ID, () => {
+          return HttpResponse.json(exampleUserAlreadyHasAClientError);
```

Comment:
This test isn't testing that it handles an api error. It's testing that when given a success response it returns whatever the response is.

We don't have error handling in setUserClientId, but we should. We need to write some error handling in setUserClientId, and then we need to response with a bad status code, so that we can test that the error handling is triggered.

Reply:
Is this handling it correctly?

```
it("handles an api failure", async () => {
  server.use(
    http.patch(AUTH0_USER_BY_ID, () => {
      return new HttpResponse(null, {
        status: 500,
      });
    }),
  );

  await expect(
    setUserClientId(exampleUserID, exampleClientID),
  ).rejects.toEqual(new Error("Internal Server Error"));
});
```

Reply:
Yep